### PR TITLE
:sparkles: Add the liveness and readiness probe in the manager deployment

### PR DESCRIPTION
**Pull request by prafull01**
**Description**
Add the liveness and readiness probe for the manager deployment.

**Motivation**
Fixes: #1761

### CODE REVIEW

File: testdata/project-v3/config/manager/manager.yaml [hunk 1]
```diff
@@ -33,6 +33,14 @@ spec:
         name: manager
         securityContext:
           allowPrivilegeEscalation: false
+        livenessProbe:
```
Comment (camilamacedo86) [inline testdata/project-v3/config/manager/manager.yaml:36]: wdyt about we add for the livenessProbe:

initialDelaySeconds: 5
      periodSeconds: 10

File: testdata/project-v3/config/manager/manager.yaml [hunk 2]
```diff
@@ -33,6 +33,14 @@ spec:
...
+          httpGet:
+            path: /readyz
+            port: 9443
+        readinessProbe:
+          httpGet:
+            path: /healthz
+            port: 9443
```
Comment (camilamacedo86) [inline testdata/project-v3/config/manager/manager.yaml:43]: And then here for the readinessProbe:

initialDelaySeconds: 5
      periodSeconds: 10

**Review by camilamacedo86 — COMMENTED** [review]
I liked this one 🥇 . its motivations came from community requests made in SDK operator-framework/operator-sdk#4102. I pinged it in the slack as well for we get feedback from the community. Let's see.

**Comment by christopherhein** [comment]
Seems pretty low risk and a good addition to level up the base scaffolding. Before #1790 is done I'll need to make sure we don't change the `9443` port and the default health/ready endpoint paths.

**Comment by camilamacedo86** [comment]
I think it is fine to get merged so;
/lgtm

**Comment by estroz** [comment]
@prafull01 did you manually test this change? I don't think these endpoints are available by default.

**Comment by prafull01** [comment]
@estroz Hi Eric, I thought the following e2e scenario would cover the checking of controller-manager coming up:
https://github.com/kubernetes-sigs/kubebuilder/blob/7db6812e44e788ba6ee1c30b41ce8cc7de8a15f6/test/e2e/v3/e2e_suite.go#L168

**Pull request merged**